src/data/bucket.js

```diff
@@ -89,6 +93,7 @@ module.exports = {
             // look up StyleLayer objects from layer ids (since we don't
             // want to waste time serializing/copying them from the worker)
             (bucket: any).layers = layers;
+            (bucket: any).stateDependentLayers = layers.filter((l) => l.isStateDependent());
```

Comment:
Pre-compute the layers with state-dependent expressions during deserialize. These are needed to re-compute the expressions in `ProgramConfiguration` until expression evaluators can be re-created when deserialized (#6255)
After that change, layer ids will be sufficient